### ISSUE TITLE
Preview 5 has two identical copies of System.Security.Cryptography.Cng.dll

### ISSUE THREAD

**Issue by ericsink** [dotnet/aspnetcore#33642]
In .NET 6 preview 5, there are identical copies of System.​Security.​Cryptography.​Cng.dll in two different targeting packs, one in Microsoft.​NETCore.​App.Ref and another in Microsoft.​AspNetCore.​App.Ref.

I first mentioned this in a Tweet:

https://twitter.com/eric_sink/status/1405727524194242566

to which @davidfowl replied "That sounds like a bug...", so I figured it should be logged, and hopefully this is the right place.

**Comment by dougbu** [comment]
Was this assembly added to Microsoft.NETCore.App.* while we weren't looking❔ We've long (at least since 3.1.0) had a weird case where Cryptography.Cng was in our targeting packs but not our runtimes (and visa-versa for Cryptography.Pkcs).

The Microsoft.NETCore.App.Ref content changed between preview 4 and 5, adding Cryptography.Cng (and Cryptograph.OpenSsl). Was that change intentional @ericstj❔ If it matters, I don't see any changes in that period between the runtimes.

**Comment by dougbu** [comment]
In any case, something odd is happening. https://github.com/dotnet/aspnetcore/blob/main/src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj#L147
in our App.Ref exclusions should mean we don't duplicate anything from Microsoft.NETCore.App.Ref in our Microsoft.AspNetCore.App.Ref.

**Comment by dougbu** [comment]
@ericstj if Cryptography.Cng was intentionally added to Microsoft.NETCore.App.Ref, the bug here is leaving System.Security.Cryptography.Cng in Microsoft.AspNetCore.Internal.Transport. Our repo expects the transport package to contain only _additional_ ref/ assemblies we need. The duplication between Microsoft.AspNetCore.Internal.Transport and Microsoft.NETCore.App.Ref can't be correct and special-casing it to avoid this bug seems **Just Wrong:tm:**

(Side note: We still don't need implementation assemblies in Microsoft.AspNetCore.Internal.Transport)

**Comment by ericstj** [comment]
Yes it was exposed with https://github.com/dotnet/runtime/pull/51853.  Cc @ViktorHofer

**Comment by ericstj** [comment]
To remove from the transport package we need to remove https://github.com/dotnet/runtime/blob/cff5854fb02259b88cab89f77e8d403809300ae8/src/libraries/System.Security.Cryptography.Cng/Directory.Build.props#L5
